cmake/SSGCommon.cmake

```diff
@@ -1004,27 +1003,39 @@ macro(ssg_build_html_stig_tables PRODUCT STIG_PROFILE DISA_STIG_VERSION)
     )
     add_dependencies(${PRODUCT}-tables generate-${PRODUCT}-table-stig)
 
-    set(SSG_LINKCHECKER_TABLE_FILE_LIST "${SSG_LINKCHECKER_TABLE_FILE_LIST};${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig.html" PARENT_SCOPE)
-    set(SSG_LINKCHECKER_TABLE_FILE_LIST "${SSG_LINKCHECKER_TABLE_FILE_LIST};${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig-manual.html" PARENT_SCOPE)
+    set(SSG_HTML_TABLE_FILE_LIST "${SSG_HTML_TABLE_FILE_LIST};${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig.html;${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig-manual.html" PARENT_SCOPE)
 
     install(FILES "${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig.html"
         DESTINATION "${SSG_TABLE_INSTALL_DIR}")
     install(FILES "${CMAKE_BINARY_DIR}/tables/table-${PRODUCT}-stig-testinfo.html"
         DESTINATION "${SSG_TABLE_INSTALL_DIR}")
 endmacro()
 
-macro(ssg_define_linkchecker_tests)
+macro(ssg_define_guide_and_table_tests)
     if (SSG_LINKCHECKER_VALIDATION_ENABLED AND LINKCHECKER_EXECUTABLE)
         add_test(
             NAME "linkchecker-ssg-guides"
-            COMMAND "${LINKCHECKER_EXECUTABLE}" --check-extern ${SSG_LINKCHECKER_GUIDE_FILE_LIST}
+            COMMAND "${LINKCHECKER_EXECUTABLE}" --check-extern ${SSG_HTML_GUIDE_FILE_LIST}
         )
 
         add_test(
             NAME "linkchecker-ssg-tables"
-            COMMAND "${LINKCHECKER_EXECUTABLE}" --check-extern ${SSG_LINKCHECKER_TABLE_FILE_LIST}
+            COMMAND "${LINKCHECKER_EXECUTABLE}" --check-extern ${SSG_HTML_TABLE_FILE_LIST}
         )
     endif()
+
+    if (GREP_EXECUTABLE)
+        foreach(TABLE_FILE ${SSG_HTML_TABLE_FILE_LIST})
+            string(REPLACE "${CMAKE_BINARY_DIR}/tables/" "" TEST_NAME "${TABLE_FILE}")
+            # -z treats newlines as regular chars so we can match multi-line
+            # -v inverts the match, we are trying to make sure the tables don't
+            #    match this pattern
+            add_test(
+                NAME "sanity-ssg-tables-${TEST_NAME}"
+                COMMAND "${GREP_EXECUTABLE}" "-zv" "</thead>[[:space:]]*</table>" "${TABLE_FILE}"
```

Comment:
It would be nice to redirect the grep output to /dev/null, now it prints the whole file on stdout, which pollutes ctest output.